# this patch was missing from https://github.com/armbian/build/blob/09e416e31cc01ece4533a65f02a470a4c21b90ea/patch/u-boot/u-boot-rockchip64/board-rock-pi-4-enable-spi-flash.patch
# and prevented booting from SPI
diff --git configs/rock-pi-4-rk3399_defconfig configs/rock-pi-4-rk3399_defconfig
index 5b779f85ec..6717bc8af6 100644
--- configs/rock-pi-4-rk3399_defconfig
+++ configs/rock-pi-4-rk3399_defconfig
@@ -12,6 +12,7 @@ CONFIG_ROCKCHIP_RK3399=y
 CONFIG_TARGET_EVB_RK3399=y
 CONFIG_DEBUG_UART_BASE=0xFF1A0000
 CONFIG_DEBUG_UART_CLOCK=24000000
+CONFIG_SPL_SPI=y
 CONFIG_SPL_SPI_FLASH_SUPPORT=y
 CONFIG_SPL_SPI_SUPPORT=y
 CONFIG_SYS_LOAD_ADDR=0x800800
diff --git configs/rock-pi-4c-rk3399_defconfig configs/rock-pi-4c-rk3399_defconfig
index f1e2d91a44..fea29a378f 100644
--- configs/rock-pi-4c-rk3399_defconfig
+++ configs/rock-pi-4c-rk3399_defconfig
@@ -12,6 +12,7 @@ CONFIG_ROCKCHIP_RK3399=y
 CONFIG_TARGET_EVB_RK3399=y
 CONFIG_DEBUG_UART_BASE=0xFF1A0000
 CONFIG_DEBUG_UART_CLOCK=24000000
+CONFIG_SPL_SPI=y
 CONFIG_SPL_SPI_FLASH_SUPPORT=y
 CONFIG_SPL_SPI_SUPPORT=y
 CONFIG_SYS_LOAD_ADDR=0x800800
